docs/sources/mimir/configure/about-ingester-adaptive-limiters.md

```diff
@@ -0,0 +1,60 @@
+---
+aliases:
+  - ../configuring/configuring-ingester-adaptive-limiters/
+description: Learn about ingester adaptive limiters.
+menuTitle: Ingester adaptive limiters
+title: About Grafana Mimir ingester adaptive limiters
+---
+
+# About Grafana Mimir ingester adaptive limiters
+
+Mimir's adaptive concurrency limiters can be used to guard against ingester overload, by automatically adapting concurreny limits based on indications of overload.
+
+## How do adaptive limiters work?
+
+Adaptive limiters detect overload by observing response times, inflight requests, and throughput.
+
+When recent response times increase significantly relative to the longer term trend, an adaptive limiter will temporarily decrease concurrency limits to avoid potential overload. Similarly, when increasing inflight requests correlate with flat or decreasing throughput and increasing response times, this is taken as a sign of overload and the concurrency limit is decreased.
+
+## How do adaptive limiters behave in practice?
+
+When an ingester is not overloaded, the concurrency limit will increase to some multiple of current inflight requests, based on the `max-limit-factor`. This provides headroom for bursts of requests without being too high to lower quickly if overload is detected.
+
+When ingester overload is detected, the limit will gradually decrease to the concurrent request processing capacity of whichever resource is overloaded, then will oscillate around that level so long as the overload continues.
+
+## Request queueing
+
+Since inflight limits are, at most, a multiple of the current inflight ingester requests, some queueing of requests will occur when an ingester's adaptive limiter is full. This allows short bursts of requests to be tolerated while still keeping the number of inflight requests under control. The amount of blocking that is allowed is based on the `initial-rejection-factor` and `max-rejection-factor`, and requests will only queue when the limiter is full.
+
+## Request rejection
+
+Ingesters support separate adaptive limiters for push and read requests since these may be overloaded by different resources at different times. When ingesters are heavily loaded these limiters may be full, and when the load is sustained, queues may fill up as well. When this happens, requests may be rejected, with a preference for rejecting read requests being before push requests.
```

Comment:
```suggestion
Ingesters support separate adaptive limiters for push and read requests, since these requests may be overloaded by different resources at different times. When ingesters are heavily loaded, these limiters may be full, and when the load is sustained, queues might fill up as well. When this happens, requests may be rejected, with a preference for rejecting read requests before rejecting push requests.
```